component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java

```diff
@@ -0,0 +1,244 @@
+                        type = DataType.STRING),
+                @Parameter(name = IBMMQConstants.PORT,
+                        description = "Port of the MQ server",
+                        type = DataType.STRING),
+                @Parameter(name = IBMMQConstants.CHANNEL,
+                        description = "Channel of the MQ server which should use to connect",
+                        type = DataType.STRING),
+                @Parameter(name = IBMMQConstants.QUEUE_MANAGER_NAME,
+                        description = "Name of the Queue Manager",
+                        type = DataType.STRING),
+                @Parameter(name = IBMMQConstants.USER_NAME,
+                        description = "User name of the server. If this is not provided, " +
+                                "will try to connect without both username and password",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "null"),
+                @Parameter(name = IBMMQConstants.PASSWORD,
+                        description = "Password of the server. If this is not provided, will try to connect without " +
+                                "both username and password",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "null"),
+        },
+        examples = {
+                @Example(description = "This example shows how to connect to an IBM MQ queue and "
+                        + "send messages.",
+                        syntax = "@source(type='ibmmq',"
+                                + "destination.name='Queue1',"
+                                + "host='192.168.56.3',"
+                                + "port='1414',"
+                                + "channel='Channel1',"
+                                + "queue.manager = 'ESBQManager',"
+                                + "password='1920',"
+                                + "username='mqm',"
+                                + "@map(type='text'))"
+                                + "define stream SweetProductionStream(name string, amount double);"),
+        }
+)
+
+public class IBMMQSink extends Sink {
+    private static final Logger LOG = LoggerFactory.getLogger(IBMMQSink.class);
+    private OptionHolder optionHolder;
+    private QueueConnection connection;
+    private MQQueueConnectionFactory connectionFactory;
+    private StreamDefinition outputStreamDefinition;
+    private QueueSession session;
+    private Queue queue;
+    private QueueSender messageSender;
+    private MessageConsumer consumer;
+    private String userName;
+    private String password;
+    private String queueName;
+    private boolean isSecured = false;
+    private SiddhiAppContext siddhiAppContext;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class, Map.class, ByteBuffer.class};
```

Comment:
do we support binary data too "ByteBuffer.class" ??

Reply:
Yes. We are checking it and sending and receiving accordingly.

Reply:
in the receiving you are not supporting  ByteBuffer.class, thats the confusion

Reply:
It was a my mistake. I changed it through 576b559df6cda24424647e6b50b3612e6dc79966